### DISCUSSION TITLE
Is it able to close only selected firefox window with "command+q"?

### DISCUSSION THREAD

**Discussion by ghost** [asmvik/yabai#1944] [Q&A] [answered]
Hello, is there anyway to have Firefox windows being separate and thus only closing the selected window with command+q rather than all open firefox windows.

Thank you.

**Comment by brorbw** [comment]
This does not really have any relevance to neither skhd or yabai and is more of a Firefox or macOS problem but allow me to answer it anywas.
`CMD+w` is a OS wide shortcut for closing the currently selected window. If the window contains tabs it will close them one by one. So for firefox and other tabbed applications you can use `CMD+SHIFT+w` to close the current window.

**Comment by Bellavene** [comment, marked as answer]
I don't know about Firefox, but in Vivaldi you can remap the cmd+Q shortcut to anything. If your program can't remap it, then you can do that with Hammerspoon, Keyboard Maestro or Bettertouchtool, etc.

**Reply by ghost** [reply]
I have been able to do this using Hammerspoon with the following config:

```
-- Define a function to close the active window
local function closeActiveWindow()
    local app = hs.application.frontmostApplication()
    local focusedWindow = app:focusedWindow()
    
    if focusedWindow then
        focusedWindow:close()
    end

    -- Check if the application has any remaining windows
    local windows = app:allWindows()
    if #windows == 0 then
        -- If there are no remaining windows, quit the application
        app:kill()
    end
end

-- Intercept CMD+Q and call the custom function
hs.hotkey.bind({"cmd"}, "q", closeActiveWindow)
```

Edit: updated to quit application if no more windows in order to fix issues with some apps (kitty and CotEditor to be precise), apps meant to be in background still stay in background
